### ISSUE TITLE
Source code

### ISSUE THREAD

**Issue by jsissom** [LsrSal/DEE_JL1#3]
Can you please publish the source code to these?

**Comment by LsrSal** [comment]
This was my solo effort to get JL1 online with GRBL. It is based on older[ 6-axis GRBL.](https://github.com/robomechs/6-AXIS-USBCNC-GRBL)
Big part of it was hand editing of HAL configuration, that usually automatically generated by Cube MX. That is not a common or practical way to develop the code.

Since then there is organized effort to get better JL1 firmware going. Please follow this: https://github.com/dlarue/STM32F1xx

**Comment by jsissom** [comment]
How can one take the 6-axis GRBL code and provide the binaries you have provided?

**Comment by jsissom** [comment]
How can one take the 6-axis GRBL code and create the binaries you have provided?

**Comment by LsrSal** [comment]
I use Notepad and Atollic. That original source for 6-axis opens and compiles in Atollic without any issues.
Most of my effort was to move Spindle/laser PWM from default port to B0, as it is tied to hardware timer TM0. I'm no software professional. It took me two weeks of reading reference literature and tracing the original code to manually edit/add a few lines to be able to move PWM to B0. My approach was to let original HAL code to finish, then to move PWM to different pin, before main application execution begins.
The rest of pins assignment is no different than Atmel version, all easily done in CPU_MAP.

While I'm not aware of any problem in my binary, I believe it is fully functional, if you like to develop new JL1 firmware I would encourage you to participate in Doug's efforts referenced in my previous message.

In Doug's version, forked from GRBLHAL, move of PWM done properly through configuration. 
Ironically, GRBLHAL for STM32F1xx is also based on 6-axis, but additional HAL layer (specific to GRBL) added for more hardware compatibility.

**Comment by jsissom** [comment]
I'm sorry, my question was poorly worded.  I was hoping you would provide the code you changed so someone can build the binaries you have distributed from source code since that is a requirement of the GPL license listed on the base code and in this repository.

**Comment by LsrSal** [comment]
I'm not organized enough to publish my code scribbles. If you are insinuating that I'm in some kind of of legal non-compliance - I will remove my work from here. I do not need any trouble. Though this would do nothing good for anyone.
I believe availability of this binary is for public benefit. I'm getting nothing in return. Well, Maybe occasional "thank you".
I have no plans to take a burden of supporting it any further or present my edits. I may if I have interest and drive and time, but I have no plans.

GPL is the first/default option when creating project on GitHub. I do not have a layer to advice me on selection of proper license.

Please tell me your purpose and intent? Are you planning to develop it further? For personal use?
Why projects I refer you to are not sufficient for you?
Or you are just trying to police GitHub and enforce license declaration?

**Comment by jsissom** [comment]
Thank you for providing this.

I loaded it into my board and wanted to make some changes.  According to the license of the code you took, that is my right, but since you are not following the license by providing your changed source code that builds the binary, I cannot do that.

I'm not the license police, as I am not the author of the original code base.  Those authors would need to decide if they want to enforce their license here.

**Comment by LsrSal** [comment]
Share the idea of improvement/change. Is this to benefit just you or more people? Maybe start the thread in Discussions. If I get interested - I might be able to help.

Pinning legal obligations to make me do stuff for free is not a good start. The way I see this license - you are free to take what I've posted and use it for any non-commercial purpose. This license does  not oblige me to perform tasks on demand. And I see no benefit to anyone to post my edits.
My point for that:

- I'm not professional SE. my code is not structured or organized to be presentable. I can see potential for more comment how clumsy it is. I do not want that.
- the way I did it is not trivial or usual or common. I expect most will get lost in my edits. Then I have to explain. I do nto have time for that. Not posting it I'm saving others from lost time.
- There are better project in progress to achieve same thin in better and organized way. It is GRBLHAL based. I think it has promise and II would encourage you to follow that.

With above, if you tell me what you like to edit - I might be able to help. For example I did compile on request custom version for T/X frame JL2 (or JL4). It was simple couple minutes task so no problem.

Let's reset this conversation:

So tell me what you intend to change or something does not work and I might be able to help.

I would like to enable second serial port and use ESP3D in SEP12E module on board. I've already flashed it. with ESP3D. I think it is more practical to allow other project to develop, then integration of second port will be simpler.
